docs/openshift/deploy.md

```diff
@@ -0,0 +1,113 @@
+
+Bonus tip: include an `--auto-suffix` option to append a randomly generated suffix to the dns-prefix to form the resource group name, for example if your workflow requires a common prefix across multiple cluster deployments. Using the `--auto-suffix` pattern appends a compressed timestamp to ensure a unique cluster name (and thus ensure that each deployment's configuration artifacts will be stored locally under a discrete `_output/<resource-group-name>/` directory).
+
+## ACS Engine the Long Way
+
+### Step 1: Generate an SSH Key
+
+In addition to using OpenShift APIs to interact with the clusters, cluster operators may access the master machine using SSH.
+
+If you don't have an SSH key [cluster operators may generate a new one](../ssh.md#ssh-key-generation).
+
+### Step 2: Create a Service Principal
+
+acs-engine needs a Service Principal to interact with Azure Resource Manager (ARM). Follow the instructions to [create a new service principal](../serviceprincipal.md).
+
+### Step 3: Edit your Cluster Definition
+
+ACS Engine consumes a cluster definition which outlines the desired shape, size, and configuration of OpenShift. There are a number of features that can be enabled through the cluster definition: check the `examples` directory for a number of... examples.
+
+Edit the [simple OpenShift cluster definition](/examples/openshift.json) and fill out the required values (every value with empty default `""` must be filled in):
+
+* `openShiftConfig`: contains authentication info for OpenShift. You can provide either `clusterUsername` and `clusterPassword` for simple htpasswd authentication, or set the `enableAADAuthentication` to `true` to enable AAD integration between OpenShift and Azure. AAD Authentication is preferred and more secure option, but requires more prerequisites and privileges
+* `azProfile`: Azure account information. `subscriptionId` and `tenantId` can be obtained from `$ az account list -o json` command. `resourceGroup` and `location` should be filled in based on your preference
+* `masterProfile`:
+  * `count`: currently only one master is supported
+  * `dnsPrefix`: must be a region-unique name and will form part of the hostname (e.g. myprod1, staging, leapingllama)
+  * `vmSize`: must be at least `Standard_D4s_v3`
+* `agentPoolProfiles`: contains specification for infrastructure and compute nodes
+  * `vmSize`: must be at least `Standard_D4s_v3`
+* `linuxProfile`: `keyData` must contain the public portion of an SSH key - this will be associated with the `adminUsername` system user (e.g. 'ssh-rsa AAAAB3NzaC1yc2EAAAADAQABA....')
+* `clientId`: this is the service principal's appId uuid or name from step 2
```

Comment:
for now, clientid and secret should be kept blank if aad integration is on.  This is because the instructions to create that service account are inadequate for aad use.